stellargraph/data/explorer.py

```diff
@@ -451,24 +435,26 @@ def transition_probability(nn):
                 current_node = node
 
                 for _ in range(length - 1):
-                    neighbours = self.graph.neighbors(
-                        current_node, use_ilocs=True, include_edge_weight=weighted
+                    # select one of the neighbours using the
+                    # appropriate transition probabilities
+                    neighbours, weights = self.graph.neighbor_arrays(
+                        current_node, include_edge_weight=True, use_ilocs=True
                     )
 
-                    if not neighbours:
+                    if len(neighbours) == 0:
                         break
 
-                    # select one of the neighbours using the
-                    # appropriate transition probabilities
-                    choice = naive_weighted_choices(
-                        rs, (transition_probability(nn) for nn in neighbours),
-                    )
+                    mask = neighbours == previous_node
+                    weights[mask] *= ip
+                    mask |= np.isin(neighbours, previous_node_neighbours)
+                    weights[~mask] *= iq
+
+                    probs = np.cumsum(weights) / weights.sum()
```

Comment:
What about moving this to `naive_weighted_choices`, so that temporal random walks see the same speedup? (I think it already has numpy array input.)

Reply:
good idea, it looks like that's the temporal random walks are the only other algorithm using this